unit_tests/RecordUnitTests.d

```diff
@@ -122,3 +122,167 @@ struct Foo
 D");
 
 }
+
+// Translate immediately declared array variable in global scope.
+unittest
+{
+    assertTranslates(
+q"C
+
+struct Bar {
+const char *foo;
+const char *oof;
+} baz[64];
+
+C",
+q"D
+extern (C):
+
+struct Bar
+{
+    const(char)* foo;
+    const(char)* oof;
+}
+
+extern __gshared Bar[64] baz;
+D");
+
+}
+
+// Do not put an extra newline after array declaration.
+unittest
+{
+    assertTranslates(q"C
+typedef int Baz;
+
+struct Foo {
+    Baz data[32];
+    char len;
+};
+C",
+q"D
+extern (C):
+
+struct Foo
+{
+    int[32] data;
```

Comment:
> I don't agree. For basic types this seems to be a major use case. Why would one otherwise use a typedef?

But how to know that in advance? The idiom to make own fixed-width integers is quite popular in C. There is no way to list all that types in code.